### PR TITLE
refactor: split `read_gbq_table` implementation into functions and move to separate module

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/session/__init__.py [hunk 1]
```diff
@@ -757,95 +701,110 @@ def _read_gbq_table(
...
-            table, api_name=api_name, use_cache=use_cache
+        # ---------------------------------
+        # Fetch table metadata and validate
+        # ---------------------------------
+
+        # TODO TODO TODO
+        # *  Validations based on index columns.
+        # 4. Create index.
+        # 5. Create ordering.
+        # TODO TODO TODO
```
Comment (tswast) [inline bigframes/session/__init__.py:725]: Ooops, these TODOs are done.

```suggestion
```

File: bigframes/session/_io/bigquery/read_gbq_table.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+        # TODO(b/335727141): If table has clustering/partitioning, fail if
+        # index_cols is empty.
+
+        # If there are primary keys defined, the query engine assumes these
+        # columns are unique, even if the constraint is not enforced. We make
+        # the same assumption and use these columns as the total ordering keys.
+        is_index_unique = len(index_cols) != 0
+    else:
+        is_index_unique = _check_index_uniqueness(
```
Comment (chelsea-lin) [inline bigframes/session/_io/bigquery/read_gbq_table.py:243]: This operation is also quiet expensive though.

**Review by chelsea-lin — APPROVED** [review]
LGTM

File: bigframes/session/_io/bigquery/read_gbq_table.py (hunk 2 above)
Comment (tswast) [reply bigframes/session/_io/bigquery/read_gbq_table.py:243]: Good point. I want to minimize any behavior changes in this PR. Filed b/338065601 to follow-up.

**Pull request merged**